cinema/views.py

```diff
@@ -56,3 +99,55 @@ def get_serializer_class(self):
             return MovieSessionDetailSerializer
 
         return MovieSessionSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset
+
+        if self.action in ("list", "retrieve"):
+            queryset = queryset.select_related(
+                "cinema_hall",
+                "movie"
+            ).annotate(
+                tickets_available=F("cinema_hall__rows")
+                * F("cinema_hall__seats_in_row")
+                - Count("tickets")
+            )
```

Comment:
It's good but I think its better for clarity to use calculated field and provide it on all layers.

Reply:
hmmm, agree, it would be more clear, thanks